### PR TITLE
Search for dead executors and throws away them and get a new ones

### PR DESCRIPTION
**Pull request by kuisathaverat**
@reviewbybees

### CODE REVIEW

**Comment by imod** [comment]
waiting for @reviewbybees ....

**Comment by Dohbedoh** [comment]
🐝

File: scriptler/restartDeadExecutors.groovy [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/*** BEGIN META {
+ "name" : "Restart Dead Executors",
+ "comment" : "Search for dead executors and throws away them and get a new ones.",
+ "core": "1.609",
+ "authors" : [
+ { name : "Kuisathaverat" }
+ ]
+ } END META**/
+
+import hudson.FilePath
```
Comment (Evildethow) [inline scriptler/restartDeadExecutors.groovy]: Unused import.

Also pre-imported anyway

File: scriptler/restartDeadExecutors.groovy [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import hudson.model.Node
+import hudson.model.Slave
```
Comment (Evildethow) [inline scriptler/restartDeadExecutors.groovy]: Unused import.

Also pre-imported anyway

File: scriptler/restartDeadExecutors.groovy [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import jenkins.model.Jenkins
```
Comment (Evildethow) [inline scriptler/restartDeadExecutors.groovy]: Pre-imported already

File: scriptler/restartDeadExecutors.groovy [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+Jenkins jenkins = Jenkins.instance
+for (Node node in jenkins.nodes) {
+  // Make sure slave is online
+  if (!node.toComputer().online) {
+    println "Node '$node.nodeName' is currently offline - skipping check"
+    continue;
```
Comment (Evildethow) [inline scriptler/restartDeadExecutors.groovy]: `continue` is unnecessary as its the last statement in the loop

File: scriptler/restartDeadExecutors.groovy [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  } else {
+    props = node.toComputer().getSystemProperties();
+    println "Node '$node.nodeName' is running ";
+    //check if has executors dead
+    for (Executor ex : node.toComputer().getExecutors()){
```
Comment (Evildethow) [inline scriptler/restartDeadExecutors.groovy]: Loop can be re-written to:

```
    node.toComputer().getExecutors()
        .findAll { it.causeOfDeath instanceof Throwable }
        .each { 
      println '[Dead]' + it.causeOfDeath
      it.doYank()
    }
```

**Comment by Evildethow** [comment]
None of the imports are necessary (pre-imported already) and can be removed if you want.

🐝

**Comment by kuisathaverat** [comment]
@reviewbybees done

File: scriptler/restartDeadExecutors.groovy (hunk 5 above)
Comment (kuisathaverat) [inline scriptler/restartDeadExecutors.groovy]: Thanks @Evildethow the next time I will do in this way

**Pull request merged**